packages/sakuli-core/src/runner/js-script-executor.class.spec.ts

```diff
@@ -136,4 +134,25 @@ describe("JsScriptExecutor", () => {
 
     expect(context.mock).toHaveBeenCalled();
   });
+
+  it("should catch rejected promise when function is not defined", async () => {
+    //GIVEN
+    const executor = new JsScriptExecutor({
+      filename: "test.js",
+      waitUntilDone: true,
+    });
+
+    //WHEN
+    const result = executor.execute(
+      stripIndents`
+        (async () => {
+             moc()
+        })()
+        `,
+      {}
+    );
+
+    //THEN
+    await expect(result).rejects.toThrowError();
```

Comment:
Minor addition: We could verify the particular error to validate it's been cause by our syntax error.